homeassistant/components/homekit/config_flow.py

```diff
@@ -542,7 +542,7 @@ async def async_step_exclude(self, user_input=None):
         entity_cat_entities = set()
         for entity_id in all_supported_entities:
             if ent_reg_ent := ent_reg.async_get(entity_id):
-                if ent_reg_ent.entity_category in ENTITY_CATEGORIES:
+                if ent_reg_ent.entity_category in EntityCategory:
```

Comment:
Do you know how python implements this under the hood?  Is it a linear search or a hash lookup.   It probably doesn't matter too much unless we get a lot of these though

Reply:
I have to admit I do not know how python implements this under the hood. Looking at the code for enum, it seems to be using :
```python
    def __contains__(cls, member):
        if not isinstance(member, Enum):
            raise TypeError(
                "unsupported operand type(s) for 'in': '%s' and '%s'" % (
                    type(member).__qualname__, cls.__class__.__qualname__))
        return isinstance(member, cls) and member._name_ in cls._member_map_
```
which points me to:
```python
    enum_class._member_map_ = {}                 # name->value map
```

Reply:
Looks like a hash lookup 👍